dashboard/src/actions/kube.tsx

```diff
@@ -146,3 +175,45 @@ export function getAndWatchResource(
     );
   };
 }
+
+// getResources requests and processes the responses for the resources
+// associated with an installed package.
+export function getResources(
+  pkg: InstalledPackageReference,
+  refs: APIResourceRef[],
+  watch: boolean,
+): ThunkAction<void, IStoreState, null, KubeAction> {
+  return dispatch => {
+    dispatch(
+      requestResources(
+        pkg,
+        refs,
+        watch,
+        (r: GetResourcesResponse) => {
+          processGetResourcesResponse(r, dispatch);
+        },
+        (e: any) => {
+          dispatch(receiveResourcesError(e));
+        },
+        (pkg: InstalledPackageReference) => {
+          dispatch(closeRequestResources(pkg));
+        },
+      ),
+    );
+  };
+}
+
+export function processGetResourcesResponse(
+  r: GetResourcesResponse,
+  dispatch: ThunkDispatch<IStoreState, null, KubeAction>,
+) {
+  const key = keyForResourceRef(
+    r.resourceRef!.apiVersion,
+    r.resourceRef!.kind,
+    r.resourceRef!.namespace,
+    r.resourceRef!.name,
```

Comment:
Are we 100% sure that `r` does have a `resourceRef` ? If not, we can just dispatch an error instead of forcing the null check? No strong opinion, though, happy as it is.